src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -170,8 +170,12 @@ public static function get_core_data() {
 			return static::$core;
 		}
 
-		$config = static::read_json_file( __DIR__ . '/theme.json' );
-		$config = static::translate( $config );
+		if ( static::theme_has_support() ) {
```

Comment:
This is not relevant for this PR but I have also found that we can remove redundant `is_readable` function calls in the `theme_has_support` method.

Currently we call it once or twice for results returned from the `get_file_path_from_theme` method: https://github.com/WordPress/wordpress-develop/blob/aff634740769140f5f9f2961ba741c544ee92f57/src/wp-includes/class-wp-theme-json-resolver.php#L608-L611

But the `get_file_path_from_theme` method suddenly also calls the same `is_readable` function before returning non-empty result: https://github.com/WordPress/wordpress-develop/blob/aff634740769140f5f9f2961ba741c544ee92f57/src/wp-includes/class-wp-theme-json-resolver.php#L633

So, we can update the `theme_has_support` method to check whether the returned result from the `get_file_path_from_theme` is not empty: 	
```diff
	public static function theme_has_support() {
		if ( ! isset( static::$theme_has_support ) ) {
			static::$theme_has_support = (
-				is_readable( static::get_file_path_from_theme( 'theme.json' ) ) ||
-				is_readable( static::get_file_path_from_theme( 'theme.json', true ) )
+				static::get_file_path_from_theme( 'theme.json' ) !== '' ||
+				static::get_file_path_from_theme( 'theme.json', true ) !== ''
			);
		}

		return static::$theme_has_support;
	}
```

P.S.: calling the `is_readable` function in the `get_file_path_from_theme` method is a controversial decision which is not expected from that function and honestly shouldn't be there IMO.

Reply:
@eugene-manuilov I've included this fix here.